### ISSUE TITLE
[Feature] Data sources for elasticsearch_roles / elasticsearch_role_mapping

### ISSUE THREAD

**Issue by matthiashaasesourceforge** [elastic/terraform-provider-elasticstack#82]
Please integrate elasticsearch_role and elasticsearch_role_mapping incl. new func... Data Sources from https://registry.terraform.io/providers/disaster37/elasticsearch

Problem:
There is a problem build our planned ELK Stack per terraform incl. role and role_mapping.
Also we need to remove not needed role and role_mapping if there are some "trojaner" added ressources (if occurs).

Ressources + Data Sources: 
elasticsearch_role and elasticsearch_role_mapping ressource needed

**Describe the solution you'd like**
migrate from disaster37 and create new one function according Data Sources

**Describe alternatives you've considered**
At moment we integrate not coming from elastic private provider "disaster37".
It's a disaster!

**Additional context**
no add context

**Comment by Grimmi2112** [comment]
I just saw that the missing "role mapping" feature was already described in Feature Request 47: https://github.com/elastic/terraform-provider-elasticstack/issues/47
It has also already been tagged for version 0.4.0, which is great. We need the feature very much!

In addition to that it would be great to have data sources for "roles" & "role mappings". With these data sources we would like to create a lister of all existing "roles" & "role mappings".
With this list we want to check in a pipeline if other roles and role mappings have been added that should not be part of the cluster. We then want to determine and delete these.
This topic is a security requirement for us to be able to prevent changes at RBAC level.
In addition, it would also be great if you could use the data sources to get the detailed information of a single "role" or a single "role mapping".

**Comment by Kushmaro** [comment]
Thanks for opening @matthiashaasesourceforge 
next time, please try to keep a sensible title as it was a bit hard to follow and understand. We will consider these for next upcoming versions.

**Comment by mikeclearbank** [comment]
we are also in need of the role mapping functionality. Its paramount for people who want to set up SSO (openid, saml etc...) for their Kibana.

**Comment by tobio** [comment]
This has been released with [0.5.0](https://github.com/elastic/terraform-provider-elasticstack/releases/tag/v0.5.0)
